### PR TITLE
Pass `localAddress` for "request" and "connect"

### PR DESCRIPTION
**Pull request by LiudasSmatavicius**

### CODE REVIEW

File: .changeset/itchy-ravens-leave.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"proxy": patch
```
Comment (TooTallNate) [inline .changeset/itchy-ravens-leave.md:2]: New feature, so this should be minor:

```suggestion
"proxy": minor
```

File: .changeset/itchy-ravens-leave.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+---
+
+Pass localAddress for request and connect
```
Comment (TooTallNate) [inline .changeset/itchy-ravens-leave.md:5]: How about something a little more descriptive, like:

```suggestion
Add `localAddress` option to specify which networking interface the proxy should use to create outgoing connections
```

**Review by TooTallNate — CHANGES_REQUESTED** [review]
I'm not seeing the `localAddress` ever get assigned anywhere, so I don't think this is working correctly as is.

**Comment by LiudasSmatavicius** [comment]
> interface ProxyServer accepts localAddress as string or undefined

**Comment by TooTallNate** [comment]
I see. An options object would be "cleaner", but ya this works.

Probably would make sense to refactor this module to use classes at some point anyways.

I'll merge this, thanks!

**Review by TooTallNate — APPROVED** [review]

**Pull request merged**